tests/phpunit/Differ/DiffColorizerTest.php

```diff
@@ -56,107 +57,136 @@ public function test_id_adds_colours_to_a_given_diff(string $originalDiff, strin
     }
 
     /**
-     * @return array<non-empty-string, list<non-empty-string>>
+     * @return Generator<non-empty-string, list<non-empty-string>>
      */
-    public static function provideDiffs(): array
+    public static function provideDiffs(): Generator
     {
-        return [
```

Comment:
I would suggest to do the array -> iterator refactoring in a separate MR because it is otherwise quite hard to read if any of the existing cases were changed